debian/NEWS

```diff
@@ -0,0 +1,36 @@
+docker.io (20.10.2-0ubuntu2) UNRELEASED; urgency=medium
+
+  As of version 20.10.2-0ubuntu2, upgrading the docker.io package will fail if
+  any containers are still using AUFS. It will be necessary to take some
+  measures to move existing containers away from aufs and delete the directory
+  /var/lib/docker/aufs before upgrading can succeed. See the previous NEWS
+  entry for instructions on moving away from AUFS.
+
+ -- William 'jawn-smith' Wilson <william.wilson@canonical.com>  Wed, 24 Feb 2021 16:06:03 -0600
+
+docker.io (1.8.3~ds1-3) unstable; urgency=medium
+
+  As of src:linux 4.0+ (specifically, >= 3.18-1~exp1), AUFS support is no longer
+  included in official Debian-compiled kernels.  What this means to Docker users
+  is that if your existing images are stored within the "AUFS" graph driver that
+  once you update your kernel, your images (and containers) will become
+  inaccessible (due to the kernel not having the necessary aufs modules to mount
+  them).  To recover from this, there are a couple options (detailed below).
+
+  1. Use the included nuke script to blow away your existing "/var/lib/docker"
+     contents and start fresh with Linux 4.0+:
+
+       service docker stop
+       /usr/share/docker.io/contrib/nuke-graph-directory.sh /var/lib/docker
+       service docker start
+
+  2. Use "docker save" (see "docker help save" for usage) before booting 4.0+ to
+     preserve your images on-disk as tar files, then follow the nuke step from
+     the previous option followed by using "docker load" to re-load your images.
+
+  3. Update to src:linux >= 4.1.1-1~exp1 ("aufs: Apply patches to enable
+     building aufs out-of-tree"), and then compile the aufs modules out-of-tree
+     (a package for doing this module compilation automatically doesn't yet
+     exist at the time of this writing, but might in the future).
```

Comment:
Hah!  Since I wrote this, `aufs-dkms` is now a thing -- is that going to be "supported" on Ubuntu to the same level it is on Debian, or are even the "AUFS enabling" patches being removed from `src:linux*` in Ubuntu?

Reply:
(If _not_, it may be worth calling that out in the above `NEWS` entry, but if so then it's probably fine to just leave as-is and make users actually dig if they want to stay on AUFS for some reason I cannot fathom.)

Reply:
Having done some more digging, `aufs-dkms` is in bad shape, and in Ubuntu is currently provided by the kernel package (and due to the basket case AUFS is, can't really be sustainable/maintainable for the wide swath of kernels Ubuntu supports), so it's probably worth making the new `NEWS` entry explicitly call out that option three here is *not* actually worth users time trying to chase down and they do need to migrate (or maintain their own kernel builds :trollface:).

Reply:
I've updated the NEWS entry. Thanks for the feedback!